pkg/build/api/v1/types.go

```diff
@@ -138,6 +138,8 @@ const (
 	BuildSourceBinary BuildSourceType = "Binary"
 	// BuildSourceImage indicates the build will accept an image as input
 	BuildSourceImage BuildSourceType = "Image"
+	// BuildSourceNone indicates the build has no predefined input (only valid for Source and Custom Strategies)
+	BuildSourceNone BuildSourceType = "None"
```

Comment:
Aren't we going to get rid of the BuildSourceType in the future?  Is this to work around validations we already have in place?